libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1390,17 +1396,37 @@ void ExpressionCompiler::appendBitOperatorCode(Token::Value _operator)
 	}
 }
 
-void ExpressionCompiler::appendShiftOperatorCode(Token::Value _operator)
+void ExpressionCompiler::appendShiftOperatorCode(Token::Value _operator, Type const& _leftType, Type const& _rightType)
 {
-	BOOST_THROW_EXCEPTION(UnimplementedFeatureError() << errinfo_comment("Shift operators not yet implemented."));
+	// stack: rvalue lvalue
+
+	IntegerType const& leftType = dynamic_cast<IntegerType const&>(_leftType);
+	bool const c_leftSigned = leftType.isSigned();
+
+	// The RValue can be a RationalNumberType too.
+	bool c_rightSigned = false;
+	if (auto rightType = dynamic_cast<RationalNumberType const*>(&_rightType))
+		c_rightSigned = rightType->integerType()->isSigned();
```

Comment:
Please add an assertion to check that `integerType()` is valid (and thus the rational number actually is an integer).